includes/Modules/Search_Console.php

```diff
@@ -246,6 +246,20 @@ protected function create_data_request( Data_Request $data ) {
 		return new WP_Error( 'invalid_datapoint', __( 'Invalid datapoint.', 'google-site-kit' ) );
 	}
 
+	/**
+	 * Returns a punycode version of a unicode URL.
+	 *
+	 * @param string $url The URL to decode.
+	 */
+	protected function decode_unicode_url( $url ) {
+		$parts = wp_parse_url( $url );
+		if ( ! $parts ) {
+			return $url;
+		}
+		$decoded = \Requests_IDNAEncoder::encode( $parts['host'] );
```

Comment:
That is probably not a big deal, but just wondering whether there's an easy way for us to detect whether that is even necessary? Also there is no case where using this on a non-unicode domain would mess up the result I suppose?

Reply:
Yes, we can call `Requests_IDNAEncoder::is_ascii first before decoding. I can add that check. Regular domains without unicode, or previously puny-encoded domains will NOT be affected by being encoded again.

Reply:
Turns out we can't call that method, it is protected... and it is already being called in the `encode` run (which calls `to_ascii`)(https://github.com/WordPress/WordPress/blob/master/wp-includes/Requests/IDNAEncoder.php#L64)